### PR TITLE
Style fixes for  Order fills page

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Reference: https://github.com/cowprotocol/explorer/pull/413#issuecomment-1491756414

## Changes
1. Made FILLED (in the header) column a little wider
2. Fixed displaying of order assets info in the top of the page (using `<TokenAmount>`)
3. Added `<TokenAmount>` component for buy/sell/execution amounts. It helps with displaying them short
4. Fixed displaying of "View batch graph" button, it was too big for mobile view
5. Fixed global layout (`MainWrapper`) to fix the page in mobile view. Warning! This change might break smth on other pages
6. Fixed execution price calculation for Fills table
7. Added control to inverse execution price 

## Background
1. `<TokenAmount>` - a new component that render amount with up to 8 decimals and show the full amount on hover of displayed one

image

image

### CODE REVIEW

File: src/components/orders/FilledProgress/index.tsx [hunk 1]
```diff
@@ -114,70 +116,39 @@ export function FilledProgress(props: Props): JSX.Element {
...
-    swappedAddress = sellTokenAddress
-    swappedAmount = executedSellAmount
-
-    // Buy orders need to add the fee, to the sellToken too (swappedAmount in this case)
-    filledAmountWithFee = filledAmount
-    swappedAmountWithFee = swappedAmount.plus(executedFeeAmount)
-  }
+  const isSell = kind === 'sell'
+
+  const mainToken = (isSell ? sellToken : buyToken) || null
```
Comment (anxolin) [inline src/components/orders/FilledProgress/index.tsx:121]: you really hate some much `let` than you added a bit of repetition. Agree on killing all those bad  lets, but why not doing a helper function with

```
if (isSell)
    return {  ... }
else 
    return {  ... }

```

File: src/components/orders/FilledProgress/index.tsx [hunk 2]
```diff
@@ -114,70 +116,39 @@ export function FilledProgress(props: Props): JSX.Element {
...
-  const FormatedText = (): JSX.Element => (
+  const OrderAssetsInfo = (): JSX.Element => (
     <>
       {' '}
-      <span>
+      <OrderAssetsInfoWrapper>
         <b>
           {/* Executed part (bought/sold tokens) */}
-          {formattedFilledAmount} {mainSymbol}
+          <TokenAmount amount={filledAmountWithFee} token={mainToken} symbol={mainSymbol}/>
```
Comment (anxolin) [inline src/components/orders/FilledProgress/index.tsx:144]: thanks for makiing the component in this project too?

File: src/components/token/TokenAmount/index.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import BigNumber from 'bignumber.js'
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+import { FormatAmountPrecision, formatSmartMaxPrecision, formattingAmountPrecision } from 'utils'
+
+// TODO: unify with TokenAmount in @cowprotocol/cowswap
```
Comment (anxolin) [inline src/components/token/TokenAmount/index.tsx:6]: great todo :)

**Review by anxolin — COMMENTED** [review]
Soft approve. Reviewed only code

I like the description, I arrived to the code with more info on what to expect to see there

I didn't check the UI yet, but if you don't have an out of scope section, it is likely you will get from Elena a lot of comments on things you were not intending to fix in this PR

File: src/components/orders/FilledProgress/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/components/orders/FilledProgress/index.tsx:121]: I just wanna have this code compact, before it took 38 lines of code, now only 12

**Comment by elena-zh** [comment]
Hey @shoom3301 , great job!
Some issues that I noticed: 
1. there are incorrect Execution time and Execution price values: I reported #424 issue for this
2. Could you please adjust the Filled column on an order details as well?
filled column
3.It would be great if you could add revert price button: 
image
Like this one 
image

Thanks!

File: src/components/orders/FilledProgress/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/components/orders/FilledProgress/index.tsx:121]: I treat it as a table:
| - | - | - | - |
|--------|--------|--------|--------|
| const mainToken = | isSell ? | sellToken | buyToken |
| const mainAddress = | isSell ? | sellTokenAddress | buyTokenAddress |

**Comment by shoom3301** [comment]
@elena-zh thanks!
Fixed all in this PR besides of execution time.
Execution time PR: https://github.com/cowprotocol/explorer/pull/425

**Comment by elena-zh** [comment]
Hey @shoom3301 , great job!
The last nitpick is to add 'Revert price' icon to the mobile card:
revert
Like here: 
image

Thank you!

File: src/components/orders/OrderDetails/FillsTable.tsx [hunk 4]
```diff
@@ -269,16 +301,16 @@ const FillsTable: React.FC<Props> = (props) => {
...
           {items.map((item, i) => (
-            <RowFill key={item.txHash} index={i + tableState.pageOffset} trade={item} />
+            <RowFill key={item.txHash} index={i + tableState.pageOffset} trade={item} isPriceInversed={isPriceInversed} />
           ))}
         </>
       )
     }
-    return tableContent
   }
+  const [isPriceInversed, setIsPriceInversed] = useState(false)
```
Comment (alfetopito) [inline src/components/orders/OrderDetails/FillsTable.tsx:313]: Typo: `inversed` -> `inverted`

**Review by alfetopito — APPROVED** [review]
Approved!

With a few suggestions for follow up PRs

1. What about moving the invert button to the `limit price` label on top?
https://user-images.githubusercontent.com/43217/230116198-0c3ada75-32ce-4f02-8ec2-f9bfcd5b9b18.mov
As you can see in the video, every time I invert the price moves around. Having it on the label will make it stick to the same place always

2. Connect the button with the whole table, inverting all prices when any is clicked

**Comment by shoom3301** [comment]
@elena-zh thanks! Added it!
@alfetopito it can be a great idea, but I don't have a strong opinion on it. Let's wait Michel and ask him :)

**Review by nenadV91 — APPROVED** [review]

File: src/components/orders/OrderDetails/FillsTable.tsx [hunk 5]
```diff
@@ -175,9 +191,31 @@ export type Props = StyledUserDetailsTableProps & {
...
+  if (!sellToken || !buyToken) return null
+
+  const sellData = { amount: sellAmount, decimals: sellToken.decimals }
+  const buyData = { amount: buyAmount, decimals: buyToken.decimals }
+
+  return calculatePrice({
+    numerator: isPriceInversed ? buyData : sellData,
+    denominator: isPriceInversed ? sellData : buyData,
+  }).multipliedBy(
+    TEN_BIG_NUMBER.exponentiatedBy((isPriceInversed ? buyToken : sellToken).decimals)
```
Comment (alfetopito) [inline src/components/orders/OrderDetails/FillsTable.tsx:214]: You still missed a few places with `inversed` :)

**Review by anxolin — APPROVED** [review]
I also think is nicer the invert in the label

It's also coherent wth the app, and not only for @alfetopito reason, but I think is easier to see in the label than in the long Limit Price description



Example where we do it in the "label" (column header in this case)
image



I think the PR grew in scope a little bit, so I would leave it as it is, and any further small change can be done in follow ups (to make the PRs more scoped)

In this line, I would suggest a follow up, where the the invertPrice state is shared between the header and body of the table of fills

See video:

https://user-images.githubusercontent.com/2352112/230353515-8ae0c4f9-e841-4828-bd92-d3f224035c64.mov

**Comment by alfetopito** [comment]
Merging to parent to consolidate.
Please address remaining comments as follow up PRs

**Comment by elena-zh** [comment]
Hey @shoom3301 , @alfetopito , @anxolin , I have created #435 task to address remaining issues in the PR (newly found by me and the one reported above).

Thanks

**Pull request merged**